package com.sonata02;

import static org.junit.Assert.assertThrows;

import org.junit.Test;



import org.junit.jupiter.api.Assertions;

public class Caltest {
	
	@Test
	public void addTest() {
		Calculator c1 = new Calculator();
		Assertions.assertEquals(4,c1.add(2, 2));
	}
		@Test
		public void mulTest() {
			Calculator c2 = new Calculator();
			Assertions.assertEquals(4,c2.mul(2, 2));
		
	

}
		@Test
		public void totalBal() {
			Bank d1 = new Bank();
			d1.balance = 234;
			Assertions.assertEquals(888,d1.deposite(654));
		}
		
		@Test
		public void totalBal1() {
			Bank d2 = new Bank();
			d2.balance = 234;
			Assertions.assertEquals(888,d2.deposite(897));
		}
		
			
			@Test
			public void totalBal2() {
				Bank w1 = new Bank();
				w1.balance = 234;
				Assertions.assertEquals(200,w1.withDraw(34));
				
				
		
		
			}
			
			@Test
			public void totalBal3() {
				Bank w2 = new Bank();
				w2.balance = 234;
				Assertions.assertEquals(200,w2.withDraw(434));
				
				
			}
			@Test
			public void divTest() {
				Object c1;
				int x =  c1.div(1,1);
				assertThrows(ArithmeticException.class,()->c1.div(1,1),"it is an error");
			}
			
			
			
			
			

	}
		




